### PR TITLE
Fix decoding for T5 models with RPE

### PR DESCRIPTION
**Pull request by MaximumEntropy**
Signed-off-by: MaximumEntropy <sandeep.subramanian.1@umontreal.ca>

# What does this PR do ?

Fixes a bug with decoding from Enc-Dec models with RPE.

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py [hunk 1]
```diff
@@ -407,12 +407,17 @@ def forward(
...
             # If enc_input is provided, we need to transpose it from [B x S x H] -> [S x B x H].
             enc_input = enc_input.transpose(0, 1)
-        elif (enc_input is None) and (enc_input_ids is not None):
+            enc_seq_length = enc_input.size(0)
+        elif enc_output is not None:
+            # If enc_output is provided in `batch_for_pipeline`, we need to transpose it from [B x S x H] -> [S x B x H].
+            enc_output = enc_output.transpose(0, 1)
+            enc_seq_length = enc_output.size(0)
+        # Only need to run encoder embedding and position ids if enc_input or enc_output is not provided.
+        else:
```
Comment (khcs) [inline nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py:419]: I wonder if this if-else is what we intended?
- if 'enc_input' is not None
- elif 'enc_output' is not None
- else ...

Would this be an edge case are not covered with this change?
- if both 'enc_input' and 'enc_output' are not None at the same time

**Review by khcs — COMMENTED** [review]
Great PR to cover the seq_length edge cases!
Left a comment wondering if we're still missing an edge-case with the if-else statement.
Thanks!

**Review by khcs — APPROVED** [review]
Looks good, thanks!

File: nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py (hunk 1 above)
Comment (MaximumEntropy) [reply nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py]: Fixed, Thanks!

**Pull request merged**